src/core/stream_recv.c

```diff
@@ -162,6 +162,77 @@ QuicStreamRecvQueueFlush(
     }
 }
 
+//
+// Processes a received RELIABLE_RESET frame's payload.
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicStreamProcessReliableResetFrame(
+    _In_ QUIC_STREAM* Stream,
+    _In_ QUIC_VAR_INT ErrorCode,
+    _In_ QUIC_VAR_INT ReliableOffset
+    )
+{
+    if (!Stream->Connection->State.ReliableResetStreamNegotiated) {
+        //
+        // The peer tried to use an exprimental feature without
+        // negotiating first. Kill the connection.
+        //
+        QuicTraceLogStreamWarning(
+            ReliableResetNotNegotiated,
+            Stream,
+            "Tried to use ReliableReset without negotiation.");
+        QuicConnTransportError(Stream->Connection, QUIC_ERROR_STREAM_STATE_ERROR);
```

Comment:
Is this the error the spec says to use?

Reply:
But I agree, QUIC_ERROR_TRANSPORT_ERROR is a better error code in our list of available error codes.